### DISCUSSION TITLE
Changing display names of boxes in GSNs

### DISCUSSION THREAD

**Discussion by MrOccam** [jonasthewolf/gsn2x#312] [Ideas]
Hi Jonas, an idea that came up recently was the issue that some organisation use the Claim, Argument, and Evidence naming convention for Goal structuring. How could gsn2x be modified to present its outputted diagrams to fit different naming conventions? perhaps this could also be implemented through stylesheets?

**Comment by jonasthewolf** [comment]
Hi @MrOccam , I tried hard with CSS yesterday, but I don't think it is possible.
What would be the difference between Claim and Argument? The top-level goal is the Claim and every other goal is Argument?
I think the easiest solution is a little Python post-processing script to change the naming.
Let me try tonight.

**Reply by MrOccam** [reply]
This might explain better than I can https://www.adelard.com/asce/cae/
Thank you for your persistence

**Reply by jonasthewolf** [reply]
Thanks for providing a reference.
However, I am not sure that it is such a simple change in gsn2x. My idea was supporting the standard.
The idea of Claim, Argument and Evidence seems related, but somehow proprietary to the Adelard tooling.
That is something I definitely don't want to compete with.

I see two options how to proceed:
a) Feel free to fork gsn2x ;-)
b) Use the following little script as a starting point to post process the generated SVG. It just renames the IDs, but does not modify the shape of the elements.

```python
import glob
import sys
import xml.etree.ElementTree as ET

for svg_file in glob.glob('**/example.gsn.svg'):
    dom = ET.parse(svg_file)
    gElems = dom.getroot().findall('.//g')
    firstGoal = True
    for gElem in gElems:
        x = gElem.findall('.//text[0]')
        if x:
            y = x[0].firstChild.nodeValue.strip()
            if y[0:2] == 'Sn':
                x[0].firstChild.nodeValue = 'Evidence' + y[2:]
            elif y[:1] == 'G':
                if firstGoal: 
                    x[0].firstChild.nodeValue = 'Claim' + y[1:]
                    firstGoal = False
                else:
                    x[0].firstChild.nodeValue = 'Argument' + y[1:]
    dom.write(sys.stdout.buffer, encoding='utf-8', xml_declaration=True)
```

**Reply by MrOccam** [reply]
Thanks Jonas. I think we'll use the python code, it would fit into the code we're using to integrate with gsn2x quite well. 

On a slightly separate note do you have any python library recommendations for running rust?

**Reply by jonasthewolf** [reply]
To my understanding, calling gsn2x is calling the same as every other binary. It has nothing to do with gsn2x being programmed in Rust.
I guess [subprocess.run](https://docs.python.org/3/library/subprocess.html#subprocess.run) seems to be the idiomatic way to call programs in Python.
